### PR TITLE
[EventHubs] geodr preview

### PR DESCRIPTION
**Pull request by swathipil**
Adding the geo replication symbol to the desired capabilities. Currently, this will only be supported by the consumer. However, adding it to producer for the future as per service team request.

No tests added for this, as we cannot create geodr resources. Just testing locally.

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/azure/eventhub/_producer.py [hunk 1]
```diff
@@ -136,6 +136,8 @@ def __init__(
     def _create_handler(
         self, auth: Union[uamqp_JWTTokenAuth, JWTTokenAuth]
     ) -> None:
+
+        desired_capabilities = [GEOREPLICATION_SYMBOL]
```
Comment (kashifkhan) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_producer.py:140]: today this is a no-op?

Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_producer.py:140]: yeah, no desired capabilities on the send side

**Review by kashifkhan — APPROVED** [review]

**Comment by swathipil** [comment]
test run: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3792144&view=results
ci run: https://dev.azure.com/azure-sdk/public/_build/results?buildId=3792143&view=results

**Pull request merged**